eng/linkerTests/linkerCorrectnessTests.targets

```diff
@@ -0,0 +1,113 @@
+<Project>
+  <PropertyGroup>
+    <LinkerTestDir>$(ArtifactsBinDir)linkerTests/</LinkerTestDir>
+    <LinkerTestProjectsDir>$(LinkerTestDir)projects/</LinkerTestProjectsDir>
+    <TestDotNetPath>$(DotNetRoot)dotnet</TestDotNetPath>
+
+    <ProjectTemplate>$(MSBuildThisFileDirectory)project.csproj.template</ProjectTemplate>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <TestConsoleAppSourceFiles Include="$(MSBuildProjectDirectory)/*.cs" />
+
+    <TestSupportFiles Include="$(MSBuildThisFileDirectory)SupportFiles/Directory.Build.props">
+      <DestinationFolder>$(LinkerTestDir)</DestinationFolder>
+    </TestSupportFiles>
+    <TestSupportFiles Include="$(MSBuildThisFileDirectory)SupportFiles/Directory.Build.targets">
+      <DestinationFolder>$(LinkerTestDir)</DestinationFolder>
+    </TestSupportFiles>
+  </ItemGroup>
+
+  <Target Name="CreateTestDir"
+          Inputs="@(TestSupportFiles)"
+          Outputs="@(TestSupportFiles->'%(DestinationFolder)\%(FileName)%(Extension)')">
+    <MakeDir Directories="%(TestSupportFiles.DestinationFolder)" />
+    <Copy SourceFiles="@(TestSupportFiles)" DestinationFolder="%(TestSupportFiles.DestinationFolder)" />
+  </Target>
+
+  <Target Name="GetTestConsoleApps">
+    <ItemGroup>
+      <TestConsoleAppSourceFiles>
+        <ProjectDir>$(LinkerTestProjectsDir)$(MSBuildProjectName)/%(Filename)/</ProjectDir>
+      </TestConsoleAppSourceFiles>
+      <TestConsoleAppSourceFiles>
+        <ProjectFile>%(ProjectDir)project.csproj</ProjectFile>
+        <AssetsFile>%(ProjectDir)obj/project.assets.json</AssetsFile>
+        <TestCommand>%(ProjectDir)bin\Release\net5.0\win-x64\publish\CoreRun.exe /v project.dll</TestCommand>
```

Comment:
Do we have to use `CoreRun.exe`? I'd prefer if we used `dotnet.exe`, since that is what a user will use.

Reply:
dotnet.exe is not placed on the self-contained app output folder, there is however a project.exe file but I believe this won't work correctly given that we are using a live-built runtime pack that doesn't know well how to build a host. I tried that initially and it was giving me wierd errors which is why I fall back to CoreRun.exe, but I'll give project.exe another shot.

Reply:
I don't think this setup is correct. The linker is primary used for runtime packs (e.g. browser-wasm) which won't work with corerun.

Reply:
I fixed this now, we are now instead just running the executable that gets produced.